storm-client/src/jvm/org/apache/storm/StormSubmitter.java

```diff
@@ -78,21 +78,18 @@ public static boolean validateZKDigestPayload(String payload) {
     @SuppressWarnings("unchecked")
     public static Map prepareZookeeperAuthentication(Map<String, Object> conf) {
         Map toRet = new HashMap();
-
+        String secretPayload = (String) conf.get(Config.STORM_ZOOKEEPER_TOPOLOGY_AUTH_PAYLOAD);
         // Is the topology ZooKeeper authentication configuration unset?
         if (! conf.containsKey(Config.STORM_ZOOKEEPER_TOPOLOGY_AUTH_PAYLOAD) ||
                 conf.get(Config.STORM_ZOOKEEPER_TOPOLOGY_AUTH_PAYLOAD) == null ||
                 !  validateZKDigestPayload((String)
                     conf.get(Config.STORM_ZOOKEEPER_TOPOLOGY_AUTH_PAYLOAD))) {
-
-            String secretPayload = generateZookeeperDigestSecretPayload();
+            secretPayload = generateZookeeperDigestSecretPayload();
+        }
             toRet.put(Config.STORM_ZOOKEEPER_TOPOLOGY_AUTH_PAYLOAD, secretPayload);
             LOG.info("Generated ZooKeeper secret payload for MD5-digest: " + secretPayload);
```

Comment:
Please move this line up to be right below line 87.  We don't want to log that we generated a payload unless we actually did.